packages/TradeIn/src/TradeInForm.tsx

```
@@ -68,17 +69,20 @@ const TradeInForm = ({
     setDropdownOptions(dropdownOptions => ({ ...dropdownOptions, ...filteredQueryResponse }));
   }, [queryResponse]);
 
-  const mapToDropdownOptions = (options: string[]) => {
-    return options.map(option => ({ value: option, label: capitaliseLabel(option) }));
+  const mapToDropdownOptions = (options: string[], isEnginePower: boolean = false) => {
+    return options.map(option => ({
+      value: option,
+      label: capitaliseLabel(isEnginePower ? convertEnginePower(option) : option)
+    }));
   };
 
-  const mapToDropdownOption = (option: string | number) => {
-    return mapToDropdownOptions([option.toString()])[0];
+  const mapToDropdownOption = (option: string | number, isEnginePower: boolean = false) => {
+    return mapToDropdownOptions([option.toString()], isEnginePower)[0];
   };
 
   const adjustValue = (prop: string, val: number | string | undefined) => {
     // If value is a number and doesn't include this fields parse it as number
-    return !isNaN(val as number) && !['phone', 'model'].includes(prop) && val
+    return !isNaN(val as number) && !['phone', 'model', 'email'].includes(prop) && val
```

1. `&& val` at the end makes no sense .. perhaps adding it to the front would be a better idea, but then again it's not even needed because `isNaN(undefined)` or `isNaN(false)` etc. would already make the condition fail early (or am i missing something);
3. It is highly unlikely that "email" would be a number .. why do we need to explicitly exclude it?

wouldn't the following be sufficient:

```
(isNaN(Number(val))) || prop === 'phone' || prop === 'model') ? val : parseInt(val);
```

unless you think `isNaN(Number(val))` for email can return true...

---

Yep, Kristina reported this issue, I also never tried to enter num, but Kristina did and didn't work, so I added this check.

---

then let's add a test here? so we can unit test this function .. and be sure about what it returns ..